toolchain/semantics/semantics_handle_if.cpp

```diff
@@ -0,0 +1,83 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/semantics/semantics_context.h"
+#include "toolchain/semantics/semantics_node.h"
+
+namespace Carbon {
+
+auto SemanticsHandleIfConditionStart(SemanticsContext& context,
+                                     ParseTree::Node parse_node) -> bool {
+  context.node_stack().Push(parse_node);
+  return true;
+}
+
+auto SemanticsHandleIfCondition(SemanticsContext& context,
+                                ParseTree::Node parse_node) -> bool {
+  // Convert the condition to `bool`.
+  auto cond_value_id = context.node_stack().Pop<SemanticsNodeId>();
+  cond_value_id = context.ImplicitAsBool(parse_node, cond_value_id);
+
+  // Create the else block and the then block, and branch to the right one. If
+  // there is no `else`, the then block will terminate with a branch to the
+  // else block, which will be reused as the resumption block.
+  auto if_block_id = context.node_block_stack().PopForAdd();
+  auto else_block_id = context.node_block_stack().PushForAdd();
+  auto then_block_id = context.node_block_stack().PushForAdd();
+
+  // Branch to the appropriate block.
+  context.AddNodeToBlock(
+      if_block_id,
+      SemanticsNode::BranchIf::Make(parse_node, then_block_id, cond_value_id));
+  context.AddNodeToBlock(
+      if_block_id, SemanticsNode::Branch::Make(parse_node, else_block_id));
+  return true;
+}
+
+auto SemanticsHandleIfStatementElse(SemanticsContext& context,
+                                    ParseTree::Node parse_node) -> bool {
+  // Switch to emitting the else block.
+  auto then_block_id = context.node_block_stack().PopForAdd();
+  context.node_stack().Push(parse_node, then_block_id);
+  return true;
+}
+
+auto SemanticsHandleIfStatement(SemanticsContext& context,
+                                ParseTree::Node parse_node) -> bool {
+  auto sub_block_id = context.node_block_stack().PopForAdd();
```

Comment:
```suggestion
  // Either the `then` or `else` block; we need to look at the next node to find out.
  auto sub_block_id = context.node_block_stack().PopForAdd();
```

Reply:
Done, but rephrased (I found "the next node" unclear in this context).